test/integration/integration.cc

```diff
@@ -186,7 +186,8 @@ void IntegrationTcpClient::write(const std::string& data, bool end_stream) {
   connection_->write(buffer, end_stream);
   do {
     connection_->dispatcher().run(Event::Dispatcher::RunType::NonBlock);
-  } while (client_write_buffer_->bytes_written() != bytes_expected);
+  } while (client_write_buffer_->bytes_written() != bytes_expected && !disconnected_);
+  EXPECT_FALSE(disconnected_);
```

Comment:
Isn't this going to fail sometimes? Or it won't fail the fuzz tests? It might be worth some comments when code is being put in just for fuzz tests? (I mostly get concerned that we will make the normal integration test code more complicated than it needs to be and more confusing...)

Reply:
I'll strengthen it a bit, but basically, we can't be disconnected and have written all the bytes out. Previously, it would just spin if we accidentally disconnected and then timeout. Fuzz tests are supposed to fail fast on disconnect here, as there's no sensible recovery.